lib/puppet/type/file/checksum.rb

```diff
@@ -33,8 +33,6 @@ def sum_stream(&block)
     "{#{type}}#{checksum}"
   end
 
-  private
-
   # Return the appropriate digest algorithm with fallbacks in case puppet defaults have not
   # been initialized.
   def digest_algorithm
```

Comment:
This was awkward, but seems necessary. I think the date checking was actually still incorrect; if you set to mtime via the digest_algorithm setting, the non-equals checking would always fail. Of course no one could ever set `digest_algorithm = mtime`, so it'd never cause a problem.